gate-proxy/src/main/kotlin/com/netflix/spinnaker/config/ProxyConfiguration.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.config
+
+import com.squareup.okhttp.OkHttpClient
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Configuration
+import java.io.File
+import java.security.KeyStore
+import javax.annotation.PostConstruct
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+
+@Configuration
+@EnableConfigurationProperties(
+  ProxyConfigurationProperties::class
+)
+open class ProxyConfiguration
+
+@ConfigurationProperties
+data class ProxyConfigurationProperties(var proxies: List<ProxyConfig> = mutableListOf()) {
+  @PostConstruct
+  fun postConstruct() {
+    for (proxy in proxies) {
+      // initialize the `okHttpClient` for each proxy
+      proxy.init()
+    }
+  }
+}
+
+data class ProxyConfig(var id: String = "",
+                       var uri: String = "",
+                       var skipHostnameVerification: Boolean = false,
+                       var keyStore: String = "",
+                       var keyStoreType: String = KeyStore.getDefaultType(),
+                       var keyStorePassword: String = "",
+                       var keyStorePasswordFile: String = "",
+                       var methods: List<String> = mutableListOf()) {
```

Comment:
I don't think empty strings is the right way to go here. `String?` makes a lot more sense to me since.

Reply:
I was (and still am) getting deserialization issues if I don't initialize ... happy to be shown secret sauce as I didn't look all that terribly deep to see where else we might be doing this in kt.